packages/components/src/components/DistributionsChart/index.tsx

```diff
@@ -204,19 +227,22 @@ const InnerDistributionsChart: FC<{
       }
 
       // samples
-      context.save();
-      context.strokeStyle = primaryColor;
-      context.lineWidth = 0.1;
-      samples.forEach((sample) => {
-        context.beginPath();
-        const x = xScale(sample);
-
-        context.beginPath();
-        context.moveTo(padding.left + x, height - sampleBarHeight);
-        context.lineTo(padding.left + x, height);
-        context.stroke();
-      });
-      context.restore();
+      if (showSamplesBar) {
+        context.save();
+        context.strokeStyle = primaryColor;
+        context.lineWidth = 0.1;
+
+        samples.forEach((sample) => {
+          context.beginPath();
+          const x = xScale(sample);
+
+          context.beginPath();
+          context.moveTo(padding.left + x, height - sampleBarHeight);
+          context.lineTo(padding.left + x, height);
+          context.stroke();
+        });
+        context.restore();
```

Comment:
In the future, we should experiment with direct [pixel manipulation](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas); drawing each line is expensive, and I suspect that drawing sample lines might be the most expensive part of rendering the chart.